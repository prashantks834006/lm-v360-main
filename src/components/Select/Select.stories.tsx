// Select.stories.ts|tsx

import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Select from './Select';

export default {
  title: 'Components/Select',
  component: Select,
} as ComponentMeta<typeof Select>;

const options = [
  {
    label: 'Apple',
    value: 'Apple',
  },
  {
    label: 'Banana',
    value: 'Banana',
  },
  {
    label: 'Orange',
    value: 'Orange',
  },
];

export const Primary: ComponentStory<typeof Select> = () => <Select options={options} title="Fruits" />;
