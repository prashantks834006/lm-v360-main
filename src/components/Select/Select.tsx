import React from 'react';
import { Select as MUISelect, FormControl, MenuItem } from '@mui/material';

interface IProps {
  options: { value: string; label: string }[];
}

const Select: React.FC<IProps> = ({ options }) => {
  return (
    <FormControl sx={{ m: 1, minWidth: 120 }} size="small">
      <MUISelect labelId="demo-select-small" id="demo-select-small">
        <MenuItem value="">
          <em>None</em>
        </MenuItem>
        {options.map(({ value, label }) => (
          <MenuItem value={value}>{label}</MenuItem>
        ))}
      </MUISelect>
    </FormControl>
  );
};

export default Select;
