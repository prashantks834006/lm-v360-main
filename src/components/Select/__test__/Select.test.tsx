import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../Select';

describe('Select', () => {
  it('should render properly', () => {
    const options = [
      {
        label: 'Apple',
        value: 'Apple',
      },
      {
        label: 'Banana',
        value: 'Banana',
      },
      {
        label: 'Orange',
        value: 'Orange',
      },
    ];

    const tree = renderer.create(<Select options={options} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
