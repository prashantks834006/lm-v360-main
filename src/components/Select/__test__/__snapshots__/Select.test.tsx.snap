// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render properly 1`] = `
<div
  className="MuiFormControl-root css-1869usk-MuiFormControl-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="demo-select-small demo-select-small"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="demo-select-small"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
